syft/tensor.py

```diff
@@ -50,6 +87,17 @@ def __isub__(self, arr_like):
         self.data = self.data - arr_like
         return self.data
 
+
+    def __dot__(self, arr_like):
+        """Returns inner product of two tensors."""
+        return syft.dot(self, arr_like)
+
+
+    def __matmul__(self, arr_like):
+        """Performs matrix multiplication between two tensors"""
+        return syft.matmul(self, arr_like)
```

Comment:
For `__dot__` and `__matmul__`, I'd run `_ensure_ndarray` on the `arr_like` to make sure the new array being passed is a numpy ndarray.